src/spikeinterface/preprocessing/normalize_scale.py

```diff
@@ -17,14 +17,15 @@ def __init__(self, parent_recording_segment, gain, offset, dtype):
         self.offset = offset
         self._dtype = dtype
 
-    def get_traces(self, start_frame, end_frame, channel_indices):
+    def get_traces(self, start_frame, end_frame, channel_indices) -> np.ndarray:
         traces = self.parent_recording_segment.get_traces(start_frame, end_frame, channel_indices)
-        scaled_traces = traces * self.gain[:, channel_indices] + self.offset[:, channel_indices]
+        traces = np.multiply(traces, self.gain[:, channel_indices], out=traces, casting="unsafe")
```

Comment:
I think we need a copy from the parent no ? If the parent is a memmap this will write to the file.
Am I wrong ?

Reply:
I think I do not like the `out=traces` because if parent is writable then we modify it and if it is not writable it will bug.

Reply:
get_tace() must create a buffer not use the one from the parent

Reply:
All right. Let's create a copy here and discuss the following outside of this PR: 

This behavior -one allocation per `get_traces`- is inconsistent across the database. To the degree that we achieve it it is incidental instead of intentional. We should make it explicit. 

To make this clear, you want: if you have a long chain of say 5 preprocessing steps and the base trace is 1 GiB like this, then spikeinterface should allocate 5 GiB in total (which hopefully would be deallocated outside of the function scope).


My own view is that the `get_trace` of  `BaseExtractors` should allocate always allocate the buffer and not use the memmap as it comes but preprocessors should not as that buffer was already safely allocated before. What do you think?

Reply:
I am 100% agree we should add the traces=None in get_traces() and this would be the unique buffer allocated.

Reply:
OK, let me add a copy here and open an issue to add this to the backlog. Good catch.

Reply:
@h-mayorquin we are checking this and this returns an `int16` if `traces` is `int16`.

The gain step should instead return floats (maybe 32) and the final step should be recasting. 

Her's a proposal:
```
scaled_traces = self.parent_recording_segment.get_traces(start_frame, end_frame, channel_indices).astype("float32", copy=True)
scaled_traces *= self.gain[:, channel_indices] # in-place
scaled_traces += self.offset[:, channel_indices] # in-place

if np.issubdtype(self._dtype, np.integer):
   scaled_traces = np.round(scaled_traces, out=scaled_traces)
return scaled_traces.astype(self._dtype, copy=False)
```

Can you check the memory consumption of this approach? Note that we have to go to float32 for the gain step!